opslevel/resource_opslevel_service.go

```diff
@@ -62,13 +62,27 @@ func newServiceResourceModel(ctx context.Context, service opslevel.Service, give
 		Language:        OptionalStringValue(service.Language),
 		LifecycleAlias:  OptionalStringValue(service.Lifecycle.Alias),
 		Name:            RequiredStringValue(service.Name),
-		Note:            givenModel.Note,
-		Owner:           OptionalStringValue(givenModel.Owner.ValueString()),
-		Parent:          OptionalStringValue(givenModel.Parent.ValueString()),
+		Note:            OptionalStringValue(service.Note),
 		Product:         OptionalStringValue(service.Product),
 		TierAlias:       OptionalStringValue(service.Tier.Alias),
 	}
 
+	if string(service.Owner.Id) == "" {
+		serviceResourceModel.Owner = types.StringNull()
+	} else if string(service.Owner.Id) == givenModel.Owner.ValueString() || service.Owner.Alias == givenModel.Owner.ValueString() {
+		serviceResourceModel.Owner = givenModel.Owner
+	} else {
+		serviceResourceModel.Owner = OptionalStringValue(service.Owner.Alias)
+	}
+
+	if service.Parent == nil {
+		serviceResourceModel.Parent = types.StringNull()
+	} else if string(service.Parent.Id) == givenModel.Parent.ValueString() || slices.Contains(service.Parent.Aliases, givenModel.Parent.ValueString()) {
+		serviceResourceModel.Parent = givenModel.Parent
+	} else {
+		serviceResourceModel.Parent = OptionalStringValue(service.Parent.Aliases[0])
```

Comment:
same comment as the owner field - i think this should fallback to the `id` rather the alias for stability reasons